irflow_client/irflow_client.py

```diff
@@ -948,37 +948,43 @@ def _get_config_args_params(self, config_args):
             protocol (str): https unless otherwise specified, default = HTTPS
             debug (bool): enable debug output, default = None
             verbose (int): turn up the verbosity default = 0 (optional)
+            proxy_user (str): the username for a proxy, default = None
+            proxy_pass (str): the password for a proxy, default = None
+            http_proxy (str): the address of the http proxy, default = None
+            https_proxy (str): the address of the https proxy, default = None
+            http_proxy_port (str): the port for the http proxy, default = None
+            https_proxy_port (str): the port for the https proxy, default = None
         """
 
         # Checking for missing config values
 
-        if isinstance(config_args['address'], str):
+        if 'address' in config_args and isinstance(config_args['address'], str):
             self.address = config_args['address']
-        elif not config_args['address']:
-            raise KeyError('You have the wrong or missing key or value')
+        elif 'address' not in config_args:
+            raise KeyError('An "address" must be submitted to establish a connection to the IR-Flow client')
         else:
-            raise KeyError('You have the wrong or missing key or value')
+            raise KeyError('The "address" submitted had an invalid value')
 
-        if isinstance(config_args['api_user'], str):
+        if 'api_user' in config_args and isinstance(config_args['api_user'], str):
             self.api_user = config_args['api_user']
-        elif not config_args['api_user']:
-            raise KeyError('You have the wrong or missing key or value')
+        elif 'api_user' not in config_args:
+            raise KeyError('An "api_user" must be submitted to establish a connection to the IR-Flow client')
         else:
-            raise KeyError('You have the wrong or missing key or value')
+            raise KeyError('The "api_user" submitted had an invalid value')
 
-        if isinstance(config_args['api_key'], str):
+        if 'api_key' in config_args and isinstance(config_args['api_key'], str):
             self.api_key = config_args['api_key']
-        elif not config_args['api_key']:
-            raise KeyError('You have the wrong or missing key or value')
+        elif 'api_key' not in config_args:
+            raise KeyError('An "api_key" must be submitted to establish a connection to the IR-Flow client')
         else:
-            raise KeyError('You have the wrong or missing key or value')
+            raise KeyError('The "api_key" submitted had an invalid value')
 
-        if config_args['protocol']:
+        if 'protocol' in config_args and isinstance(config_args['protocol'], str):
             self.protocol = config_args['protocol']
         else:
             self.protocol = 'https'
-        if config_args['debug']:
-            self.debug = config_args['debug']
+        if 'debug' in config_args and config_args['debug']:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The if statement can be replaced with 'var = bool(test)'](https://app.codacy.com/app/Syncurity/irflow-sdk-python/pullRequest?prid=3256097)